cosmos/config.py

```diff
@@ -0,0 +1,122 @@
+"""Module that contains all Cosmos config classes."""
+
+from __future__ import annotations
+
+import shutil
+from dataclasses import dataclass, field
+from pathlib import Path
+from logging import getLogger
+
+from cosmos.constants import TestBehavior, ExecutionMode, LoadMode
+from cosmos.exceptions import CosmosValueError
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class RenderConfig:
+    """
+    Class for setting general Cosmos config.
+
+    :param emit_datasets: If enabled test nodes emit Airflow Datasets for downstream cross-DAG
+    dependencies. Defaults to True
+    :param test_behavior: The behavior for running tests. Defaults to after each
+    :param load_method: The parsing method for loading the dbt model. Defaults to AUTOMATIC
+    :param select: A list of dbt select arguments (e.g. 'config.materialized:incremental')
+    :param exclude: A list of dbt exclude arguments (e.g. 'tag:nightly')
+    """
+
+    emit_datasets: bool = True
+    test_behavior: TestBehavior = TestBehavior.AFTER_EACH
+    load_method: LoadMode = LoadMode.AUTOMATIC
+    select: list[str] = field(default_factory=list)
+    exclude: list[str] = field(default_factory=list)
+
+
+@dataclass
+class ProjectConfig:
+    """
+    Class for setting project config.
+
+    :param dbt_project_path: The path to the dbt project directory. Example: /path/to/dbt/project
+    :param models_dir: The path to the dbt models directory within the project. Defaults to models
+    :param seeds_dir: The path to the dbt seeds directory within the project. Defaults to seeds
+    :param snapshots_dir: The path to the dbt snapshots directory within the project. Defaults to
+    snapshots
+    :param manifest: The path to the dbt manifest file. Defaults to None
+    """
+
+    dbt_project: str | Path
+    models: str | Path = "models"
+    seeds: str | Path = "seeds"
+    snapshots: str | Path = "snapshots"
+    manifest: str | Path | None = None
+
+    manifest_path: Path | None = None
+
+    def __post_init__(self) -> None:
+        "Converts paths to `Path` objects."
+        self.dbt_project_path = Path(self.dbt_project)
+        self.models_path = self.dbt_project_path / Path(self.models)
+        self.seeds_path = self.dbt_project_path / Path(self.seeds)
+        self.snapshots_path = self.dbt_project_path / Path(self.snapshots)
```

Comment:
Following the [discussion](https://github.com/astronomer/astronomer-cosmos/pull/385/files#r1272817393) from the previous PR:

I believe we should:
(a) document that models/seeds/snapshots must be relative to the `dbt` project
(b) validate if they are valid, raising an exception if they don't exist

Reply:
clarified the name so that there's no ambiguity